### ISSUE TITLE
unetbootin fails with lubuntu-16.10-alternate-i386.iso

### ISSUE THREAD

**Issue by mvsfrasson** [unetbootin/unetbootin#95]
Hi

Unetbootin failed with iso file lubuntu-16.10-alternate-i386.iso.  I verified md5sum of the iso file and it matches the hash provided at http://cdimage.ubuntu.com/lubuntu/releases/16.10/release/MD5SUMS.

I first reproduced the bug with UNetbootin from ubuntu repositories, version 608(*). To report this bug, I took the step of download source, compile myself with instructions provided and reproduced the bug with version 625.

(* by the way, I only managed to get version number from synaptic! Manpage is empty, no -v option, passing option sends messages from gksu, a mess...)

Same flashdisk succeeds with other image lubuntu-16.10-desktop-i386.iso (not alternate), and same flashdisk AND lubuntu-16.10-alternate-i386.iso succeeds with usb-creator-gtk.  Tried format with several flavors of fat (tried gparted and disks), same bug. Tried another flashdisk, same bug.

With unetbootin 625 and lubuntu-16.10-alternate-i386.iso, it freezes at 26%, 503 or 504 files out of 1894, at some files at pool/main/b/bind9/, curiously changing filename but apparently always at file 504 (maybe 503) out of 1894.

What is happening?

Best regards

Miguel

**Comment by mvsfrasson** [comment]
I realized that lubuntu-16.10-desktop-i386.iso has just 317 files in I use UNetbootin, while alternate iso image has 1894. Bug happens at file 504 of 1894.

A guess. Maybe some filelist of length 512 inside UNetbootin gets full with 504 files and some folders.

**Comment by gkovacs** [comment]
correct, 7-zip (used for extracting the ISO files) has a filesize length limit of 512 characters and unfortunately truncates them. I reported it 6 years ago to https://sourceforge.net/p/sevenzip/bugs/1055/ and still hasn't been fixed :( will probably need to switch to something else as fixing ISO support doesn't seem to be a priority for 7-zip

**Comment by ysalmon** [comment]
I think I stumbled into this bug with KUbuntu 20.04.

Three files get their name truncated :
```
./pool/restricted/l/linux-restricted-modules/linux-modules-nvidia-390-generic-hwe-20.04_5.4.0-26.30+2_amd64.deb
./pool/restricted/l/linux-restricted-modules/linux-modules-nvidia-440-generic-hwe-20.04_5.4.0-26.30+2_amd64.deb
./pool/restricted/n/nvidia-graphics-drivers-440/xserver-xorg-video-nvidia-440_440.82+really.440.64-0ubuntu6_amd64.deb
```
Since KUbuntu 20.04 livecd self-tests during boot, it shows an error.

It would be nice of UNetBootin would use the md5sum.txt file to run an integrity check and maybe work around this 7zup bug/limitation by renaming the files based on their md5 sum. Or at least show a warning about the missing files and an explanation, so that users can correct this themselves.
